cmake/CMakeLists.txt

```diff
@@ -823,7 +823,12 @@ else()
   list(APPEND onnxruntime_EXTERNAL_LIBRARIES ${CMAKE_DL_LIBS} Threads::Threads)
 endif()
 
-
+# Version parts for Windows.AI.MachineLearning.dll and onnxruntime.dll
+set(WINML_VERSION_MAJOR_PART   0 CACHE STRING "First part of numeric file/product version.")
+set(WINML_VERSION_MINOR_PART   0 CACHE STRING "Second part of numeric file/product version.")
+set(WINML_VERSION_BUILD_PART   0 CACHE STRING "Third part of numeric file/product version.")
+set(WINML_VERSION_PRIVATE_PART 0 CACHE STRING "Fourth part of numeric file/product version.")
+set(WINML_VERSION_STRING       "Internal Build" CACHE STRING "String representation of file/product version.")
```

Comment:
We shouldn't have `WINML_VERSION_*` configurations when WinML is disabled... Maybe rename to onnxruntime instead of WINML?

Reply:
beat me to it!

---
In reply to: [375561884](https://github.com/microsoft/onnxruntime/pull/2978#discussion_r375561884) [](ancestors = 375561884)

Reply:
I was trying to differentiate it from the other ORT_VERSION variable that is extracted from the VERSION_NUMBER file but not used for populating the ORT metadata. How about just VERSION_*?

Reply:
Ended up specifying ORT or WINML for variables that are specific to one or the other, and include no prefix for shared variables